compile/compile.go

```diff
@@ -302,21 +366,27 @@ func (c *Compiler) init() error {
 		c.entrypointrefs = append(c.entrypointrefs, ast.NewTerm(r))
 	}
 
-	if c.optimizationLevel > 0 && len(c.entrypointrefs) == 0 {
-		return errors.New("bundle optimizations require at least one entrypoint")
-	}
+	return nil
+}
 
-	switch c.target {
-	case TargetWasm:
-		if len(c.entrypointrefs) == 0 {
-			return errors.New("wasm compilation requires at least one entrypoint")
+// Once the bundle has been loaded, we can check the entrypoint counts.
+func (c *Compiler) checkNumEntrypoints(afterBundleLoad bool) error {
+	if !c.useRegoAnnotationEntrypoints || afterBundleLoad {
+		if c.optimizationLevel > 0 && len(c.entrypointrefs) == 0 {
+			return errors.New("bundle optimizations require at least one entrypoint")
 		}
-	case TargetPlan:
-		if len(c.entrypointrefs) == 0 {
-			return errors.New("plan compilation requires at least one entrypoint")
+
+		switch c.target {
```

Comment:
Can we make that `if c.target != TargetRego`? We don't need the `wasm/plan` in the error message, I think. Or we replace that with `c.target`, whatever it is.

Reply:
This was just a straight port of the logic that was already there--which generated per-target error messages. I like the idea of parametrizing the error message though. I'll roll that in with my next batch of changes.